group/views.py

```diff
@@ -68,4 +71,40 @@ def manage(request, group_id):
 def all(request):
     return render(request, "group/all.html", {
         'groups': Group.objects.all(),
-    })
\ No newline at end of file
+    })
+
+@login_required
+def join_group(request, group_id):
+    group = get_object_or_404(Group, pk=group_id)
+    if (request.user in group.members.all()):
+        group.members.remove(request.user)
+        joined = False
+    else:
+        group.members.add(request.user)
+        joined = True
+    
+    data = {
+        'joined': joined,
+        'member_count': group.members.count()
+    }
+
+    return JsonResponse(data)
+
+@login_required
+def create_group_request(request, group_id):
```

Comment:
same as up there, name could be "handle_group_request" or "create_or_delete_group_request"